app/org/sagebionetworks/bridge/BridgeUtils.java

```diff
@@ -280,4 +282,18 @@ public static DateTime getDateTimeOrDefault(String value, DateTime defaultValue)
             map.remove(key);
         }
     }
+    
+    public static String encodeURIComponent(String component) {
+        String encoded = null;
+        if (component != null) {
+            try {
+                encoded = URLEncoder.encode(component, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                // UTF-8 is always supported, so this should never happen. 
+                throw new BadRequestException(e.getMessage());
```

Comment:
Would a BridgeServiceException make more sense? BadRequestException suggests the client did something wrong